src/main/java/com/google/edith/ShelfDataReader.java

```diff
@@ -0,0 +1,102 @@
+package com.google.edith;
+
+import com.google.common.base.Splitter;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Processes data file of product expiration information to populate future user grocery lists. */
+public class ShelfDataReader {
+  private static final String NO_EXPIRATION = "NO_EXPIRATION";
+
+  /** Finds the specified product in the file. */
+  public static String readFile(String itemName) {
+    URL jsonresource = ShelfDataReader.class.getClassLoader().getResource("foodkeeper.json");
+    File shelfLifeData = new File(jsonresource.getFile());
+
+    JsonParser jsonParser = new JsonParser();
+    List<JsonArray> potentialMatches = new ArrayList<JsonArray>();
+    try (FileReader reader = new FileReader(shelfLifeData)) {
+      JsonObject data = (JsonObject) jsonParser.parseReader(reader).getAsJsonObject();
+      JsonArray sheets = data.getAsJsonArray("sheets");
+      // Get(2) is used because the product data starts at that index, previous data is all
+      // header data on the file.
+      JsonObject productList = sheets.get(2).getAsJsonObject();
+      JsonArray productListData = productList.getAsJsonArray("data");
+
+      for (int i = 0; i < productListData.size(); i++) {
+        JsonArray product = productListData.get(i).getAsJsonArray();
+        JsonObject nameObject = product.get(2).getAsJsonObject();
+        String productName = nameObject.get("Name").getAsString();
+        if (productName.toLowerCase().contains(itemName.toLowerCase())) {
```

Comment:
Should we add a comment regarding the upcoming NLP changes. "This algo currently has some false positives, but should be addressed with NLP changes".. along those lines